lib/active_model/serializer.rb

```diff
@@ -168,18 +147,6 @@ def json_key
       root || object.class.model_name.to_s.underscore
     end
 
-    def attributes
-      attributes = self.class._attributes.dup
-
-      attributes.each_with_object({}) do |name, hash|
-        if self.class._fragmented
-          hash[name] = self.class._fragmented.public_send(name)
-        else
-          hash[name] = send(name)
-        end
-      end
-    end
-
```

Comment:
I personally don't think associations are any less fundamental to serializers than attributes. Plus extracting semantically delimited code to a module does not make it less fundamental in my opinion.